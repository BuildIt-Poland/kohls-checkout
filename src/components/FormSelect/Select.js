import styled from 'styled-components';

import SelectBase from '../Select';

const Select = styled(SelectBase)`
  margin-top: 0.4rem;
`;

export default Select;
