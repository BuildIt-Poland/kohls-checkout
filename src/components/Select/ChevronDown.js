import styled from 'styled-components';

import Chevron from '../Icons/Chevron';

const ChevronDown = styled(Chevron)`
  position: absolute;
  top: 10px;
  right: 0;
  margin-right: 1rem;
`;

export default ChevronDown;
