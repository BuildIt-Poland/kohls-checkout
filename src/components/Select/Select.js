import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Chevron from '../Icons/Chevron';

import {
  borderRadiusLarge,
  spacingSmall,
  borderWidthThin,
  colorBlack,
  colorRed,
  colorWhite,
  colorLightGrey
} from '../../styles/designTokens';

const SelectContainer = styled.div`
  position: relative;
`;

const StyledChevronDown = styled(Chevron)`
  position: absolute;
  top: 1.3rem;
  right: 0;
  margin-right: 1rem;
`;

const StyledSelect = styled.select`
  appearance: none;
  border: ${borderWidthThin} solid ${props => (props.error ? colorRed : colorBlack)};
  border-radius: ${borderRadiusLarge};
  background-color: ${colorWhite};
  padding: ${spacingSmall};
  outline: 0;
  width: 100%;

  :disabled {
    background-color: ${colorLightGrey};
  }
`;

const Select = function({ className, label, error, type, options, ...props }) {
  return (
    <SelectContainer className={className}>
      <StyledChevronDown disabled={props.disabled} fill={colorBlack} />
      <StyledSelect error={error} {...props}>
        {options.map((props, index) => (
          <option key={index} {...props} />
        ))}
      </StyledSelect>
    </SelectContainer>
  );
};

Select.propTypes = {
  error: PropTypes.bool,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.number.isRequired,
      disabled: PropTypes.bool
    })
  )
};

export default Select;
