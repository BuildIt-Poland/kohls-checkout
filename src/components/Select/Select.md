```js
const props = {
  options: [
    {
      label: 'Alabama',
      value: 'AL'
    },
    {
      label: 'Hawaii',
      value: 'HI'
    },
    {
      label: 'Oklahoma',
      value: 'OK'
    }
  ]
};

<>
  <Select {...props} />
  <Select {...props} disabled />
  <Select {...props} error />
</>;
```
