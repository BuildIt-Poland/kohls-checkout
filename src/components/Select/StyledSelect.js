import styled from 'styled-components';

import {
  borderRadiusLarge,
  spacingSmall,
  borderWidthThin,
  colorBlack,
  colorRed,
  colorWhite,
  colorLightGrey
} from '../../styles/designTokens';

const StyledSelect = styled.select`
  appearance: none;
  border: ${borderWidthThin} solid ${props => (props.error ? colorRed : colorBlack)};
  border-radius: ${borderRadiusLarge};
  background-color: ${colorWhite};
  padding: ${spacingSmall};
  outline: 0;
  width: 100%;

  :disabled {
    background-color: ${colorLightGrey};
  }
`;

export default StyledSelect;
