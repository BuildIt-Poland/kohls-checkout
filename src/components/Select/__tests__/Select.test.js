import React from 'react';
import { create } from 'react-test-renderer';

import Select from '../Select';

jest.mock('../../Icons/Chevron', () => 'Chevron');

const options = [
  {
    label: '1',
    value: '1'
  },
  {
    label: '2',
    value: '2'
  }
];

describe('Select', () => {
  describe('When Select is enabled', () => {
    it('Renders correctly the component', () => {
      const component = create(<Select value="MM" options={options} />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('When Select is disabled', () => {
    it('Renders correctly the component', () => {
      const component = create(<Select value="MM" disabled="true" options={options} />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('When Select as an error', () => {
    it('Renders correctly the component', () => {
      const component = create(<Select value="MM" error="true" options={options} />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
