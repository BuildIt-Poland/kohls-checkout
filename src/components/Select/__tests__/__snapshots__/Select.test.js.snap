// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select When Select as an error Renders correctly the component 1`] = `
.c1 {
  position: absolute;
  top: 10px;
  right: 0;
  margin-right: 1rem;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #e0151d;
  border-radius: 0;
  background-color: #ffffff;
  padding: 0 1rem;
  outline: 0;
  width: 100%;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.25);
  height: 4rem;
}

.c2:disabled {
  background-color: #b0b0b0;
}

<div
  className="c0"
>
  <Chevron
    className="c1"
    fill="#222222"
  />
  <select
    className="c2"
  >
    <option
      label="MM"
      value={null}
    />
    <option
      label="1"
      value="1"
    />
    <option
      label="2"
      value="2"
    />
  </select>
</div>
`;

exports[`Select When Select is disabled Renders correctly the component 1`] = `
.c1 {
  position: absolute;
  top: 10px;
  right: 0;
  margin-right: 1rem;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #b0b0b0;
  border-radius: 0;
  background-color: #ffffff;
  padding: 0 1rem;
  outline: 0;
  width: 100%;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.25);
  height: 4rem;
}

.c2:disabled {
  background-color: #b0b0b0;
}

<div
  className="c0"
>
  <Chevron
    className="c1"
    disabled={true}
    fill="#222222"
  />
  <select
    className="c2"
    disabled={true}
  >
    <option
      label="MM"
      value={null}
    />
    <option
      label="1"
      value="1"
    />
    <option
      label="2"
      value="2"
    />
  </select>
</div>
`;

exports[`Select When Select is enabled Renders correctly the component 1`] = `
.c1 {
  position: absolute;
  top: 10px;
  right: 0;
  margin-right: 1rem;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #e0151d;
  border-radius: 0;
  background-color: #ffffff;
  padding: 0 1rem;
  outline: 0;
  width: 100%;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.25);
  height: 4rem;
}

.c2:disabled {
  background-color: #b0b0b0;
}

<div
  className="c0"
>
  <Chevron
    className="c1"
    disabled={false}
    fill="#222222"
  />
  <select
    className="c2"
    disabled={false}
  >
    <option
      label="MM"
      value={null}
    />
    <option
      label="1"
      value="1"
    />
    <option
      label="2"
      value="2"
    />
  </select>
</div>
`;
